core/src/main/java/de/danoeh/antennapod/core/feed/FeedMedia.java

```diff
@@ -579,4 +580,12 @@ private void checkEmbeddedPicture() {
             hasEmbeddedPicture = Boolean.FALSE;
         }
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof RemoteMedia) {
```

Comment:
The id field comes simply from the database, so it should be unique among the same type of FeedComponent. On the other hand, since the same episode would have a different id across devices, we naturally need a more meaningful way to check if they match.